source/frontend-docs.rst

```diff
@@ -0,0 +1,218 @@
+Front-end documentation with JSDoc
+##################################
+
+Writing maintainable and understandable front-end JavaScript code calls for writing
+particularly thorough documentation. JavaScript is an unusually flexible language in which
+protocols and expectations are rarely if ever obvious without carefully reading
+the code. It's also a language with a huge and diverse set of libraries where
+standards and conventions are in a constant state of flux. The only way to
+ensure that the developer who inherits your front-end code tomorrow or a year
+from now will understand what your program does is to carefully document your code,
+more so than you may be accustomed to doing on Python and other non-JS projects.
+
+A good way to document your JS is by using `JSDoc <http://usejsdoc.org/>`_ and its
+commenting conventions. JSDoc is an API documentation generator that can be used
+to generate a complete documentation site from comments in your JS code. The
+comment syntax associated with JSDoc, derived from that of JavaDoc, is also widely
+used in major JS projects. Whether or not you use JSDoc to generate an API
+docs site for your code, following the JSDoc conventions is a good way to ensure
+that your comments are as informative and immediately comprehensible as possible.
+
+JSDoc installation and basic setup
+==================================
+
+JSDoc can be added to an existing project as an npm dependency. To take advantage
+of JSDoc 3.4's native support for ECMAScript 2015, install a version of JSDoc
+later than 3.4.0::
+
+    npm install jsdoc^3.4.0 --save-dev
+
+For most projects, you will want to create a configuration file that will
+tell JSDoc such things as which files to ignore, which plugins to use, etc.
+Full instructions can be found `on the JSDoc site <http://usejsdoc.org/about-configuring-jsdoc.html>`_.
+
+One important step is to `create a configuration file <http://usejsdoc.org/about-configuring-jsdoc.html#configuration-file>`_.
+Our recommended practice is to call it ``.jsdoc.json`` and to place it in your root
+project directory (alongside other config files like ``.gitignore``, ``.babelrc``, etc).
+This configuration file specifies the plugins JSDoc will use, among other things.
+
+In your config file, you can set up the source and destination directories
+that JSDoc will look through as it crawls your JS code and builds the documentation
+site. In projects created with our project template, the source dir will look
+like ``project_name/static/js``; we recommend ``docs/js`` as the output dir.
+You'll also want to enable the ``recurse`` option so that your entire source tree
+will be scanned. If your JavaScript build output is located in the same directory
+as your source, you will also want to throw in an ``excludePattern``. Including
```

Comment:
oh - this is referring to babelifying the javascript, not creating docs from it.  I was thinking about building documentation when I read "JavaScript build output" - maybe we could be a bit more explicit here about which build we're referring to?